pkg/downloader/chart_downloader.go

```diff
@@ -125,6 +137,34 @@ func (c *ChartDownloader) DownloadTo(ref, version, dest string) (string, *proven
 	return destfile, ver, nil
 }
 
+// https://github.com/kubernetes/helm/issues/2937
+// tries to fix URL because of BUG #2937
+// transforms URL from
+// https://url.to.repo/path?key=value/chart-0.1.0.tgz
```

Comment:
How would we have gotten this broken url in the first place? What produces the broken url? Can that be fixed instead?

Reply:
Removed the method and found the place where corrupt URL is constructed